File: src/main/java/scripting/portal/PortalScriptMethods.java
package scripting.portal;

import client.Client;
import field.object.portal.FieldPortal;
import scripting.AbstractPlayerInteraction;

public class PortalScriptMethods extends AbstractPlayerInteraction {

    private final FieldPortal portal;

    public FieldPortal getPortal() {
        return portal;
    }

    public PortalScriptMethods(Client c, FieldPortal portal) {
        super(c);
        this.portal = portal;
    }

    public void enter() {
        portal.forceEnter(c.getCharacter());
    }
}

File: src/main/java/scripting/portal/PortalScriptMethods.kt
package scripting.portal

import client.Client
import field.obj.portal.FieldPortal
import scripting.AbstractPlayerInteraction

@Deprecated("Old")
class PortalScriptMethods(c: Client, private val portal: FieldPortal) : AbstractPlayerInteraction(c) {

    fun getPortal(): FieldPortal {
        return portal
    }

    fun enter() {
        portal.forceEnter(c.character)
    }
}